src/scrap.py

```diff
@@ -0,0 +1,262 @@
+import requests
+import os
+import discord
+from typing import List
+from bs4 import BeautifulSoup
+from dotenv import load_dotenv
+import sqlite3
+
+# For using my .ENV files
+load_dotenv()
+
+
+# -------------------FOR SQLLITE----------------------------
+
+# https://docs.python.org/3/library/sqlite3.html
+# https://www.blog.pythonlibrary.org/2021/09/30/sqlite/#:~:text=Here%20is%20how%20you%20would%20create%20a%20SQLite,the%20sqlite3%20module%20will%20create%20an%20empty%20database.
+
+
+def create():
+    # Creates the DB. Only needs to be called once.
+
+    connection = sqlite3.connect("jobs.db")
+    cursor = connection.cursor()
+
+    cursor.execute(
+        """CREATE TABLE IF NOT EXISTS jobs(_company TEXT, _title TEXT, _location TEXT, _link TEXT, _processed BOOLEAN DEFAULT 0)"""
+    )
+
+    connection.commit()
+    connection.close()
+
+
+def ingest_opportunities(job_data):
+    # Inserts opportunities if and only if they do not already exist
+
+    connection = sqlite3.connect("jobs.db")
+    cursor = connection.cursor()
+
+    for job in job_data:
+        cursor.execute(
+            "SELECT COUNT(*) FROM jobs WHERE _company = ?, AND _title = ?, AND _location = ?",
+            (job["_company"], job["_title"], job["_location"]),
+        )
+        counter = cursor.fetchone()[0]
+
+        # https://pynative.com/python-cursor-fetchall-fetchmany-fetchone-to-read-rows-from-table/
+
+        if counter == 0:  # The Job Exists if the counter is 0
+            cursor.execute(
+                "INSERT INTO jobs VALUES (?, ?, ?, ?, ?)",
+                (job["_company"], job["_title"], job["_location"], job["_link"], False),
+            )
+        else:
+            continue
+
+    connection.commit()
+    connection.close()
+
+
+# -------------------FOR REDDIT----------------------------
+
+
+# TO-DO: Fix this up to add the company and the location
+def reddit_scrape() -> List[object]:
```

Comment:
function names should be consistent: `reddit_scrape()`, `rapid_response()`, `indeed_response()` ?